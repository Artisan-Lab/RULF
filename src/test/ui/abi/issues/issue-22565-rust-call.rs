#![feature(unboxed_closures)]

extern "rust-call" fn b(_i: i32) {}
//~^ ERROR functions with the "rust-call" ABI must take a single non-self argument that is a tuple

trait Tr {
    extern "rust-call" fn a();

    extern "rust-call" fn b() {}
    //~^ ERROR functions with the "rust-call" ABI must take a single non-self argument
}

struct Foo;

impl Foo {
    extern "rust-call" fn bar() {}
    //~^ ERROR functions with the "rust-call" ABI must take a single non-self argument
}

impl Tr for Foo {
    extern "rust-call" fn a() {}
    //~^ ERROR functions with the "rust-call" ABI must take a single non-self argument
}

fn main () {
    b(10);

    Foo::bar();

    <Foo as Tr>::a();
    <Foo as Tr>::b();
}
