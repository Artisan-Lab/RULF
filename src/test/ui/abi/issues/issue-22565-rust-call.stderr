error: functions with the "rust-call" ABI must take a single non-self argument that is a tuple
  --> $DIR/issue-22565-rust-call.rs:3:1
   |
LL | extern "rust-call" fn b(_i: i32) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: functions with the "rust-call" ABI must take a single non-self argument that is a tuple
  --> $DIR/issue-22565-rust-call.rs:9:5
   |
LL |     extern "rust-call" fn b() {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: functions with the "rust-call" ABI must take a single non-self argument that is a tuple
  --> $DIR/issue-22565-rust-call.rs:16:5
   |
LL |     extern "rust-call" fn bar() {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: functions with the "rust-call" ABI must take a single non-self argument that is a tuple
  --> $DIR/issue-22565-rust-call.rs:21:5
   |
LL |     extern "rust-call" fn a() {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

